Analysis/EventFiltering/PWGUD/diffractionSelectors.h

```diff
@@ -0,0 +1,184 @@
+    //            command line options! This is not what we want!
+    LOGF(info, "Track types");
+    for (auto& track : tracks) {
+      LOGF(info, "  : %i / %i / %i / %i", track.trackType(), track.isGlobalTrack(), track.isGlobalTrackSDD(), track.itsClusterMap());
+    }
+
+    // tracks with TOF hits
+    // ATTENTION: in Run 2 converted data hasTOF is often true although the tofSignal
+    //            has a dummy value (99998.) This will be corrected in new versions
+    //            of the data - check this!
+    LOGF(info, "TOF");
+    int nTracksWithTOFHit = 0;
+    for (auto& track : tracks) {
+      LOGF(info, "  [%i] signal %f / chi2: %f", track.hasTOF(), track.tofSignal(), track.tofChi2());
+    }
+    LOGF(info, "Tracks with TOF hit %i", nTracksWithTOFHit);
+
+    // only tracks with good PID
+    for (auto& track : tracks) {
+      auto goodPID = hasGoodPID(diffCuts, track);
+    }
+
+    // Decisions which need past-future protection
+    // This applies to 'slow' detectors
+    // loop over all selected BCs
+    LOGF(info, "Number of close BCs: %i", bcs.size());
+    for (auto& bc : bcs) {
+      LOGF(info, "BC %i / %i", bc.globalBC(), bc.triggerMask());
+
+      // check no activity in FT0
+      LOGF(info, "FT0 %i", collision.foundFT0());
+      if (bc.has_ft0()) {
+        LOGF(info, "  %f / %f / %f / %f",
+             bc.ft0().timeA(), bc.ft0().amplitudeA()[0],
+             bc.ft0().timeC(), bc.ft0().amplitudeC()[0]);
+      }
+
+      // check no activity in FV0-A
+      LOGF(info, "FV0-A");
+      if (bc.has_fv0a()) {
+        LOGF(info, "  %f / %f", bc.fv0a().time(), bc.fv0a().amplitude()[0]);
+      }
+
+      // check no activity in FV0-C
+      // ATTENTION: not available in Run 3 data
+      //LOGF(info, "FV0-C");
+      //if (bc.has_fv0c()) {
+      //  LOGF(info, "  %f / %f", bc.fv0c().time(), bc.fv0c().amplitude()[0]);
+      //}
+
+      // check no activity in FDD (AD Run 2)
+      LOGF(info, "FDD");
+      if (bc.has_fdd()) {
+        LOGF(info, "  %f / %f / %f / %f",
+             bc.fdd().timeA(), bc.fdd().amplitudeA()[0],
+             bc.fdd().timeC(), bc.fdd().amplitudeC()[0]);
+      }
+
+      // check no activity in ZDC
+      LOGF(info, "ZDC");
+      if (bc.has_zdc()) {
+        LOGF(info, "  %f / %f / %f / %f",
+             bc.zdc().timeZEM1(), bc.zdc().energyZEM1(),
+             bc.zdc().timeZEM2(), bc.zdc().energyZEM2());
+      }
+
+      // check no activity in muon arm
+      LOGF(info, "Muons %i", muons.size());
+      for (auto& muon : muons) {
+        LOGF(info, "  %i / %f / %f / %f", muon.trackType(), muon.eta(), muon.pt(), muon.p());
+      }
+    }
+
+    // ATTENTION: currently all events are selected
+    return true;
+  };
+};
+
+// function to check if track provides good PID information
+// Checks the nSigma for any particle assumption to be within limits.
+template <typename TC>
+bool hasGoodPID(cutHolder diffCuts, TC track)
```

Comment:
Here i see you apply the PID cuts on single tracks, while if i am not wrong, in the Run-2 UD analyses 2-track selections were used. I am not asking anything specifically, but if 2-track selections will be applied than probably this function an maybe the cutHolder object might need some changes.

Reply:
There are interesting cases with more than 2 tracks. However I might introduce an additional function which checks for PID combinations, like e.g. accept 4pi events but not 3pi+1e.